pkg/trojan/message_test.go

```diff
@@ -0,0 +1,248 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package trojan
+
+import (
+	"bytes"
+	"encoding/binary"
+	"math/big"
+	"reflect"
+	"testing"
+
+	chunktesting "github.com/ethersphere/bee/pkg/storage/testing"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+// arbitrary targets for tests
+var t1 = Target([]byte{57, 120})
+var t2 = Target([]byte{209, 156})
+var t3 = Target([]byte{156, 38})
+var t4 = Target([]byte{89, 19})
+var t5 = Target([]byte{22, 129})
+var testTargets = Targets([]Target{t1, t2, t3, t4, t5})
+
+// arbitrary topic for tests
+var testTopic = NewTopic("foo")
+
+// newTestMessage creates an arbitrary Message for tests
+func newTestMessage(t *testing.T) Message {
+	payload := []byte("foopayload")
+	m, err := NewMessage(testTopic, payload)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return m
+}
+
+// TestNewMessage tests the correct and incorrect creation of a Message struct
+func TestNewMessage(t *testing.T) {
+	smallPayload := make([]byte, 32)
+	m, err := NewMessage(testTopic, smallPayload)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// verify topic
+	if m.Topic != testTopic {
+		t.Fatalf("expected message topic to be %v but is %v instead", testTopic, m.Topic)
+	}
+
+	maxPayload := make([]byte, MaxPayloadSize)
+	if _, err := NewMessage(testTopic, maxPayload); err != nil {
+		t.Fatal(err)
+	}
+
+	// the creation should fail if the payload is too big
+	invalidPayload := make([]byte, MaxPayloadSize+1)
+	if _, err := NewMessage(testTopic, invalidPayload); err != ErrPayloadTooBig {
+		t.Fatalf("expected error when creating message of invalid payload size to be %q, but got %v", ErrPayloadTooBig, err)
+	}
+}
+
+// TestWrap tests the creation of a chunk from a list of targets
+// its address length and span should be correct
+// its resulting address should have a prefix which matches one of the given targets
+// its resulting data should have a hash that matches its address exactly
+func TestWrap(t *testing.T) {
+	m := newTestMessage(t)
+	c, err := m.Wrap(testTargets)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	addr := c.Address()
+	addrLen := len(addr.Bytes())
+	if addrLen != swarm.HashSize {
+		t.Fatalf("chunk has an unexpected address length of %d rather than %d", addrLen, swarm.HashSize)
+	}
+
+	addrPrefix := addr.Bytes()[:len(testTargets[0])]
+	if !contains(testTargets, addrPrefix) {
+		t.Fatal("chunk address prefix does not match any of the targets")
+	}
+
+	data := c.Data()
+	dataSize := len(data)
+	expectedSize := 8 + swarm.ChunkSize // span + payload
+	if dataSize != expectedSize {
+		t.Fatalf("chunk data has an unexpected size of %d rather than %d", dataSize, expectedSize)
+	}
+
+	span := binary.LittleEndian.Uint64(data[:8])
+	remDataLen := len(data[8:])
```

Comment:
`rem`?

Reply:
`aining` 

(probably renaming this would be a good idea)